src/services/importTracker.ts

```diff
@@ -471,9 +475,11 @@ export function findModuleReferences(program: Program, sourceFiles: readonly Sou
             }
         }
 
+        const jsxImport = getJSXRuntimeImport(getJSXImplicitImportBase(compilerOptions, referencingFile), compilerOptions);
         forEachImport(referencingFile, (_importDecl, moduleSpecifier) => {
             const moduleSymbol = checker.getSymbolAtLocation(moduleSpecifier);
-            if (moduleSymbol === searchModuleSymbol) {
+
+            if (moduleSymbol === searchModuleSymbol && !(moduleSpecifier.text === jsxImport && nodeIsSynthesized(moduleSpecifier))) {

```

Comment:
You don't need to specifically check for the `jsxImport`, any node for which `nodeIsSynthesized` is true is going to be a hassle for the import tracker, since it has no associated positions to report. Just checking that will also remove any errors related to finding all references on the synthetic `tslib` import we also add. However, just skipping the result leaves a bit lacking, no? That's why the other PR adds a new reference type and actually records the file as a reference for the import.